include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -223,10 +255,21 @@ class fm_index
      * No guarantees.
      */
     void construct(text_t const & text)
+        //!\cond
+        requires !is_collection
+        //!\endcond
     {
          // text must not be empty
-        if (text.begin() == text.end())
+        if (std::ranges::begin(text) == std::ranges::end(text))
```

Comment:
you can do just begin and end

Reply:
why is that?

Reply:
(include seqan3/range/shortcuts.hpp)

Reply:
So, I can either do
`seqan3::begin`
`std::ranges::begin` or
`ranges::begin`

just `begin` can't be resolved
```
error: no matching function for call to 'seqan3::fm_index<std::vector<std::vector<seqan3::dna4> >, seqan3::fm_index_default_traits>::begin(const std::vector<std::vector<seqan3::dna4> >&)'
        if (begin(text) == end(text))
note: candidate: seqan3::fm_index<text_t, fm_index_traits>::cursor_type seqan3::fm_index<text_t, fm_index_traits>::begin() const [with text_t = std::vector<std::vector<seqan3::dna4> >; fm_index_traits = seqan3::fm_index_default_traits; seqan3::fm_index<text_t, fm_index_traits>::cursor_type = seqan3::fm_index_cursor<seqan3::fm_index<std::vector<std::vector<seqan3::dna4> >, seqan3::fm_index_default_traits> >]
        cursor_type begin() const noexcept
```